### PR TITLE
Provide Kubeflow PaddleJob support for MultiKueue

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
The PR introduces a new MultiKueue adapter to handle PaddleJob (Kubeflow).
We want to extend MultiKueue capabilities to satisfy the needs of early adopters.

#### Which issue(s) this PR fixes:

Relates #2552

#### Does this PR introduce a user-facing change?

```release-note
MultiKueue: Support for the Kubeflow PaddleJob
```

### CODE REVIEW

File: pkg/controller/jobs/kubeflow/jobs/paddlejob/paddlejob_multikueue_adapter.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	remoteJob = &kftraining.PaddleJob{
+		ObjectMeta: api.CloneObjectMetaForCreation(&localJob.ObjectMeta),
+		Spec:       *localJob.Spec.DeepCopy(),
+	}
+
+	// add the prebuilt workload
+	if remoteJob.Labels == nil {
+		remoteJob.Labels = map[string]string{}
```
Comment (tenzen-y) [inline pkg/controller/jobs/kubeflow/jobs/paddlejob/paddlejob_multikueue_adapter.go:70]:
```suggestion
		remoteJob.Labels = make(map[string]string, 2)
```

File: pkg/controller/jobs/kubeflow/jobs/paddlejob/paddlejob_multikueue_adapter.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		return client.IgnoreNotFound(err)
+	}
+	return client.IgnoreNotFound(remoteClient.Delete(ctx, &job))
+}
+
+func (b *multikueueAdapter) KeepAdmissionCheckPending() bool {
+	return false
+}
+
+func (b *multikueueAdapter) IsJobManagedByKueue(ctx context.Context, c client.Client, key types.NamespacedName) (bool, string, error) {
```
Comment (tenzen-y) [inline pkg/controller/jobs/kubeflow/jobs/paddlejob/paddlejob_multikueue_adapter.go:91]:
```suggestion
func (b *multikueueAdapter) IsJobManagedByKueue(context.Context, client.Client, types.NamespacedName) (bool, string, error) {
```

**Review by tenzen-y — COMMENTED** [review]
overall lgtm
just some nits.

**Comment by mszadkow** [comment]
@tenzen-y updated ptal

**Review by tenzen-y — COMMENTED** [review]
Thanks
/lgtm
/approve

**Pull request merged**